MediaBrowser.Common/Plugins/PluginManifest.cs

```diff
@@ -0,0 +1,94 @@
+#nullable enable
+using System;
+using System.Text.Json.Serialization;
+using MediaBrowser.Common.Json.Converters;
+using MediaBrowser.Model.Plugins;
+
+namespace MediaBrowser.Common.Plugins
+{
+    /// <summary>
+    /// Defines a Plugin manifest file.
+    /// </summary>
+    public class PluginManifest
+    {
+        /// <summary>
+        /// Gets or sets the category of the plugin.
+        /// </summary>
+        [JsonPropertyName("category")]
+        public string Category { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the changelog information.
+        /// </summary>
+        [JsonPropertyName("changelog")]
+        public string Changelog { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the description of the plugin.
+        /// </summary>
+        [JsonPropertyName("description")]
+        public string Description { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the Global Unique Identifier for the plugin.
+        /// </summary>
+        [JsonPropertyName("guid")]
+        [JsonConverter(typeof(JsonGuidDashConverter))]
```

Comment:
```suggestion
        [JsonConverter(typeof(JsonConverter<Guid>))]
```

Untested, but can probably use the built-in converter, as serializing the guid with dashes is how the default converter works

Reply:
I used the JsonGuidDashConvertor as i was getting this with the default.
```
[2020-12-22 09:10:03.963 +00:00] [ERR] [1] Emby.Server.Implementations.Plugins.PluginManager: Error deserializing "C:\ProgramData\Jellyfin\Server\plugins\Bookshelf_5.0.0.0".
System.InvalidOperationException: The converter specified on 'MediaBrowser.Common.Plugins.PluginManifest.Id' does not derive from JsonConverter or have a public parameterless constructor.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_SerializationConverterOnAttributeInvalid(Type classType, MemberInfo memberInfo)
   at System.Text.Json.JsonSerializerOptions.GetConverterFromAttribute(JsonConverterAttribute converterAttribute, Type typeToConvert, Type classTypeAttributeIsOn, MemberInfo memberInfo)
   at System.Text.Json.JsonSerializerOptions.DetermineConverter(Type parentClassType, Type runtimePropertyType, MemberInfo memberInfo)
   at System.Text.Json.JsonClassInfo.GetConverter(Type type, Type parentClassType, MemberInfo memberInfo, Type& runtimeType, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo.AddProperty(MemberInfo memberInfo, Type memberType, Type parentClassType, Nullable`1 parentTypeNumberHandling, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo.CacheMember(Type declaringType, Type memberType, MemberInfo memberInfo, Nullable`1 typeNumberHandling, Dictionary`2 cache, Dictionary`2& ignoredMembers)
   at System.Text.Json.JsonClassInfo..ctor(Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializerOptions.GetOrAddClass(Type type)
   at System.Text.Json.JsonSerializerOptions.GetOrAddClassForRootType(Type type)
   at System.Text.Json.ReadStack.Initialize(Type type, JsonSerializerOptions options, Boolean supportContinuation)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](Utf8JsonReader& reader, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at Emby.Server.Implementations.Plugins.PluginManager.LoadManifest(String dir) in C:\Code\jellyfin\Emby.Server.Implementations\Plugins\PluginManager.cs:line 491
```

Reply:
The default also removes the - from the guid on write.

Reply:
Use JsonGuidDashConverter due to the error below.

Tried adding a parameterless constructor to the manifest class - but still getting the error.

```
System.InvalidOperationException: The converter specified on 'MediaBrowser.Common.Plugins.PluginManifest.Id' does not derive from JsonConverter or have a public parameterless constructor.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_SerializationConverterOnAttributeInvalid(Type classType, MemberInfo memberInfo)
   at System.Text.Json.JsonSerializerOptions.GetConverterFromAttribute(JsonConverterAttribute converterAttribute, Type typeToConvert, Type classTypeAttributeIsOn, MemberInfo memberInfo)
   at System.Text.Json.JsonSerializerOptions.DetermineConverter(Type parentClassType, Type runtimePropertyType, MemberInfo memberInfo)
   at System.Text.Json.JsonClassInfo.GetConverter(Type type, Type parentClassType, MemberInfo memberInfo, Type& runtimeType, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo.AddProperty(MemberInfo memberInfo, Type memberType, Type parentClassType, Nullable`1 parentTypeNumberHandling, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo.CacheMember(Type declaringType, Type memberType, MemberInfo memberInfo, Nullable`1 typeNumberHandling, Dictionary`2 cache, Dictionary`2& ignoredMembers)
   at System.Text.Json.JsonClassInfo..ctor(Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializerOptions.GetOrAddClass(Type type)
   at System.Text.Json.JsonSerializerOptions.GetOrAddClassForRootType(Type type)
   at System.Text.Json.WriteStack.Initialize(Type type, JsonSerializerOptions options, Boolean supportContinuation)
   at System.Text.Json.JsonSerializer.WriteCore[TValue](Utf8JsonWriter writer, TValue& value, Type inputType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize[TValue](TValue& value, Type inputType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize[TValue](TValue value, JsonSerializerOptions options)
   at Emby.Server.Implementations.Plugins.PluginManager.SaveManifest(PluginManifest manifest, String path) in C:\Code\jellyfin\Emby.Server.Implementations\Plugins\PluginManager.cs:line 311
 ```

Reply:
`[JsonConverter(typeof(JsonConverterFactory))]` works in my testing

Reply:
Just uploaded a version that works without a converter. Attributes removed as well.


```
_jsonOptions = JsonDefaults.GetOptions();
            _jsonOptions.WriteIndented = true;

            // We need to use the default GUID converter, so we need to remove any custom ones.
            for (int a = _jsonOptions.Converters.Count - 1; a >= 0; a--)
            {
                if (_jsonOptions.Converters[a] is JsonGuidConverter convertor)
                {
                    _jsonOptions.Converters.Remove(convertor);
                    break;
                }
            }
```

Reply:
Did think about just doing a hack and removing the 1st convertor, but this way around, there isn't any dependency upon JsonDefaults.GetOptions being in a specific order.